### PR TITLE
tabnet

### PR DESCRIPTION
**Pull request by javaThonc**
updating Tabnet model and its performance on Alpha 158

### CODE REVIEW

File: examples/benchmarks/README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ The numbers shown below demonstrate the performance of the entire `workflow` of
 | XGBoost (Tianqi Chen, et al.) | Alpha158 | 0.0481±0.00 | 0.3659±0.00| 0.0495±0.00 | 0.4033±0.00 | 0.1111±0.00 | 1.2915±0.00| -0.0893±0.00 |
 | LightGBM (Guolin Ke, et al.) | Alpha158 | 0.0475±0.00 | 0.3979±0.00| 0.0485±0.00 | 0.4123±0.00 | 0.1143±0.00 | 1.2744±0.00| -0.0800±0.00 |
 | MLP | Alpha158 | 0.0358±0.00 | 0.2738±0.03| 0.0425±0.00 | 0.3221±0.01 | 0.0836±0.02 | 1.0323±0.25| -0.1127±0.02 |
+| TabNet with pretrain (Sercan O. Arikm et al) | Alpha158 | 0.0344±0.00|0.205±0.11|0.0398±0.00 |0.3479±0.01|0.0827±0.02|1.1141±0.32 |-0.0925±0.02  |
```
Comment (you-n-g) [inline examples/benchmarks/README.md:28]: How many times did you train the model to get this result?

File: qlib/contrib/model/pytorch_tabnet.py [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+import torch.optim as optim
+import torch.nn.functional as F
+from torch.autograd import Function
+
+from ...model.base import Model
+from ...data.dataset import DatasetH
+from ...data.dataset.handler import DataHandlerLP
+
+
+class TabNet_Model(Model):
```
Comment (you-n-g) [inline qlib/contrib/model/pytorch_tabnet.py:31]: Under in class is not the preferred naming rules
https://github.com/amontalenti/elements-of-python-style#preferred-naming-rules

**Comment by you-n-g** [comment]
@javaThonc 
Please sign the Contributor License Agreement. 
image

File: examples/benchmarks/README.md (hunk 1 above)
Comment (javaThonc) [reply examples/benchmarks/README.md:28]: I trained it three times with different random seeds

File: qlib/contrib/model/pytorch_tabnet.py (hunk 2 above)
Comment (javaThonc) [reply qlib/contrib/model/pytorch_tabnet.py]: fixed!

**Review by Gam3356 — COMMENTED** [review]
Thank you guys! Willie Parish

**Pull request merged**